### PR TITLE
Add a nativescript zone implementation

### PR DESCRIPTION
**Pull request by hdeshev**
The [nativescript-angular](https://github.com/NativeScript/nativescript-angular/) mobile renderer was using the zone-node.js bundle until 0.6.18+ added patching for the `timers` module and `fetch` promise-likes. Those are incompatible with NativeScript and cause runtime errors. The best solution to support zone.js in NativeScript in a way that minimizes future breakage seems to involve creating a separate bundle.

I based this bundle on the node one, just removing the unsupported patches. I also added a rudimentary test task that runs under node (to avoid spinning up device emulators and complicate build infrastructure).

`nativescript-angular` vendors the `zone-nativescript.js` bundle that this PR adds, but it would be great if we get rid of that in the future.

This PR takes care of #437 and the `fetch()` change is somewhat related to #439.

### CODE REVIEW

**Comment by hdeshev** [comment]
I signed it!

File: dist/zone-nativescript.js [hunk 1]
```diff
@@ -0,0 +1,842 @@
+var Zone$1 = (function (global) {
```
Comment (mhevery) [inline dist/zone-nativescript.js]: Please Remove the changes to the `dist` folder. We update the dist folder on releases only.

File: gulpfile.js [hunk 2]
```diff
@@ -169,3 +174,31 @@ gulp.task('test/node', ['compile'], function(cb) {
   jrunner.addSpecFiles(specFiles);
   jrunner.execute();
 });
+
+gulp.task('test/nativescript', ['compile'], function(cb) {
```
Comment (mhevery) [inline gulpfile.js]: Please also update the `package.json` file so that native scripts can be run using `npm run test-nativescript` command.

File: lib/zone.ts [hunk 3]
```diff
@@ -1148,7 +1148,7 @@ const Zone: ZoneType = (function(global: any) {
       const fetchPromise = global['fetch']();
       // ignore output to prevent error;
       fetchPromise.then(() => null, () => null);
-      if (fetchPromise.constructor != NativePromise) {
+      if (fetchPromise.constructor != NativePromise && fetchPromise.constructor != ZoneAwarePromise) {
```
Comment (mhevery) [inline lib/zone.ts]: Can you break this change into a separate commit. I don't think it has anything to do with NativeScript

File: test/nativescript_entry_point.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+import '../lib/zone-spec/sync-test';
+import '../lib/zone-spec/task-tracking';
+import '../lib/zone-spec/wtf';
+
+// Setup test environment
+import './test-env-setup';
+
+// List all tests here:
+import './common_tests';
+import './nativescript_tests';
```
Comment (mhevery) [inline test/nativescript_entry_point.ts]: `nativescript_tests.ts` seems to be an empty file. Was that intended?

File: dist/zone-nativescript.js (hunk 1 above)
Comment (hdeshev) [inline dist/zone-nativescript.js]: Done.

File: gulpfile.js (hunk 2 above)
Comment (hdeshev) [inline gulpfile.js]: Done

File: lib/zone.ts (hunk 3 above)
Comment (hdeshev) [inline lib/zone.ts]: Already done. The PR contains both commits.

File: test/nativescript_entry_point.ts (hunk 4 above)
Comment (hdeshev) [inline test/nativescript_entry_point.ts]: I thought it would look consistent with the node_tests.ts setup, but I can delete it. Your call.
